src/middleware/index.ts

```diff
@@ -109,71 +196,93 @@ export const validateProvider = async (
  */
 export const validateCollection = async (
   req: Request,
-  res: Response,
+  _res: Response,
   next: NextFunction
 ) => {
-  const { providerId, collectionId } = req.params;
+  const {
+    headers,
+    params: { providerId, collectionId },
+  } = req;
   const query = { provider: providerId, conceptId: collectionId };
 
   try {
     const {
       items: [collection],
-    } = await getCollections(query, { headers: req.headers });
+    } = await getCollections(query, { headers });
 
-    if (!collection) {
-      return res.status(404).json({
-        errors: `Collection with ID [${collectionId}] in provider [${providerId}] not found.`,
-      });
-    }
+    if (!collection)
+      return next(
+        new ItemNotFound(
+          `Collection with ID [${collectionId}] in provider [${providerId}] not found.`
+        )
+      );
 
     req.collection = collection;
-
-    next();
   } catch (err) {
-    console.error("A problem occurred querying for collections.", err);
-    return res.status(503).json(ERRORS.serviceUnavailable);
+    console.error(
+      "A problem occurred querying for collections.",
+      (err as Error).stack
+    );
+    return next(new ServiceUnavailableError(ERRORS.serviceUnavailable));
   }
+  next();
 };
 
 const inclusiveBetween = (v: number, mmin: number, mmax: number) =>
   mmin <= v && v <= mmax;
+
 const validLat = (lat: number) => inclusiveBetween(lat, -90.0, 90.0);
+
 const validLon = (lon: number) => inclusiveBetween(lon, -180.0, 180.0);
-const validBbox = (bbox: string) => {
-  // 2D bounding box only, CMR does not support 3D bounding box '(lon, lat, elevation)'
-  const parsedBbox = parseOrdinateString(bbox);
-  if (parsedBbox.length !== 4) return false;
 
-  const [swLon, swLat, neLon, neLat] = parsedBbox;
+const validBbox = (bbox: string | number[]) => {
+  const parsedBbox =
+    typeof bbox === "string" ? parseOrdinateString(bbox) : bbox;
+
+  if (parsedBbox.length !== 4 && parsedBbox.length !== 6) return false;
+
+  let swLon, swLat, neLon, neLat;
+  if (parsedBbox.length === 4) {
+    [swLon, swLat, neLon, neLat] = parsedBbox;
+  } else {
+    [swLon, swLat, , neLon, neLat] = parsedBbox;
+  }
   return (
-    validLon(swLon) && validLat(swLat) && validLon(neLon) && validLat(neLat)
+    validLon(swLon) &&
+    validLat(swLat) &&
+    validLon(neLon) &&
+    validLat(neLat) &&
+    swLat <= neLat &&
+    swLon <= neLon
   );
 };
 
-const validateStacOrThrow = (query: StacQuery) => {
-  const limit = query.limit ?? 10;
+const validateQueryTerms = (query: StacQuery) => {
+  const { bbox, intersects, datetime, limit: strLimit } = query;
+
+  const limit = Number.isNaN(Number(strLimit)) ? null : Number(strLimit);
```

Comment:
I don't think this can happen but, I had an issue once caused by `Number('')` resolving to 0 instead of NaN